src/model/image-tag.ts

```diff
@@ -0,0 +1,124 @@
+import Platform from './platform';
+
+class ImageTag {
+  public repository: string;
+  public name: string;
+  public version: string;
+  public platform: any;
+  public builderPlatform: string;
+  public customImage: any;
+
+  constructor(unityVersion: string, customImage: any) {
+    if (!ImageTag.versionPattern.test(unityVersion)) {
+      throw new Error(`Invalid version "${unityVersion}".`);
+    }
+
+    const builderPlatform = ImageTag.getTargetPlatformToImageSuffixMap(
+      Platform.types.StandaloneLinux64,
+      unityVersion,
+    );
+
+    this.repository = 'unityci';
+    this.name = 'editor';
+    this.version = unityVersion;
+    this.platform = Platform.types.StandaloneLinux64;
+    this.builderPlatform = builderPlatform;
+    this.customImage = customImage;
+  }
+
+  static get versionPattern() {
+    return /^20\d{2}\.\d\.\w{3,4}|3$/;
+  }
+
+  static get imageSuffixes() {
+    return {
+      generic: '',
+      webgl: 'webgl',
+      mac: 'mac-mono',
+      windows: 'windows-mono',
+      linux: 'base',
+      linuxIl2cpp: 'linux-il2cpp',
+      android: 'android',
+      ios: 'ios',
+      facebook: 'facebook',
+    };
+  }
+
+  static getTargetPlatformToImageSuffixMap(platform, version) {
+    const { generic, webgl, mac, windows, linux, linuxIl2cpp, android, ios, facebook } =
+      ImageTag.imageSuffixes;
+
+    const [major, minor] = version.split('.').map((digit) => Number(digit));
+    // @see: https://docs.unity3d.com/ScriptReference/BuildTarget.html
+    switch (platform) {
+      case Platform.types.StandaloneOSX:
+        return mac;
+      case Platform.types.StandaloneWindows:
+        return windows;
+      case Platform.types.StandaloneWindows64:
+        return windows;
+      case Platform.types.StandaloneLinux64: {
+        // Unity versions before 2019.3 do not support il2cpp
+        if (major >= 2020 || (major === 2019 && minor >= 3)) {
+          return linuxIl2cpp;
+        }
+        return linux;
+      }
+      case Platform.types.iOS:
+        return ios;
+      case Platform.types.Android:
+        return android;
+      case Platform.types.WebGL:
+        return webgl;
+      case Platform.types.WSAPlayer:
+        return windows;
+      case Platform.types.PS4:
```

Comment:
I'm wondering why we're returning `windows` for these instead of explicitly letting the user know some of these are unsupported. I assume this was already the case and no matter what, I think we should go forward with typescript so I don't expect a fix for this. Can always be done in a future pr too.

Reply:
Currently, `windows` maps to `windows-mono`, so we'll definitely need to make some change here when the windows images with IL2CPP are ready, but yes, that should be in a future PR.